### PR TITLE
[SPEED-784] New CI github action and fix existing SA errors in guardrail

### PR DESCRIPTION
**Pull request by bknutson123**
This PR does the following:
* Adds CI pipeline with github action that will run all unit tests, static index and static analyze
* Removes unused variables
* fixes several bugs found from running static analysis. Most of which are trivial. Here are a few descriptions of the more complex ones
          * changed up the merge true asserts on leave method and made it only run for the BooleanOp expression node. This method will now merge whatever is set on the left/right side of the expression into the parent scope.
          * Changed BooleanOr to call the same method as mentioned above
          * BooleanAnd also needed to pop and push the assertsTrue scope so that the right side of the BooleanOp expressions had access to previous assertions.
          * In the ScopeVar class we make sure that the two variable types are set to the same before merging the type. If they are not the same then we cannot for certain say what the variable type is.
          * Fix sql for the SqlLiteSymbolTable class so that running guardrail in the CI action works
          * Make the getChainedPropertyFetchName work when there is an ArrayDimFetch expression as part of the fetch
          * Make getScopeClone return the Scope implementation rather than the PluginScopeInterface to make SA happy with types.

### CODE REVIEW

File: src/Config.php [hunk 1]
```diff
@@ -327,7 +327,7 @@ private function parseArgv(array $argv) {
 						throw new InvalidConfigException;
 					}
 					++$argCount;
-					list($wholeMatch, $this->partitionNumber, $this->partitions) = $params;
```
Comment (jongardiner) [inline src/Config.php:330]: Why is this necessary?  We never use the variable.

File: src/Phases/IndexingPhase.php [hunk 2]
```diff
@@ -267,7 +265,7 @@ public function indexString(string $name, string $fileData): int
 				$this->traverser1->traverse($statements);
 				$this->traverser2->traverse($statements);
 			}
-		} catch (\Exception $exc) {
+		} catch (Throwable $exc) {
```
Comment (jongardiner) [inline src/Phases/IndexingPhase.php:268]: Why not exception here?

File: src/Config.php (hunk 1 above)
Comment (bknutson123) [reply src/Config.php:330]: yeah we removed it because it was not being used
